core/jobs/batch_jobs/manual_voice_artist_names_job.py

```diff
@@ -0,0 +1,504 @@
+# assume that PTransform class is of type Any. Thus to avoid MyPy's error
+# (Class cannot subclass 'PTransform' (has type 'Any')), we added an
+# ignore here.
+class FilterCommitLogModelByExplorationIds(beam.DoFn): # type: ignore[misc]
+    """DoFn to filter exploration commit log entry models based on curated
+    exploration IDs.
+    """
+
+    def process(
+        self,
+        exp_commit_log_model: exp_models.ExplorationCommitLogEntryModel,
+        exploration_ids: List[str]
+    ) -> Iterable[exp_models.ExplorationCommitLogEntryModel]:
+        if exp_commit_log_model.exploration_id in exploration_ids:
+            yield exp_commit_log_model
+
+
+# TODO(#15613): Here we use MyPy ignore because the incomplete typing of
+# apache_beam library and absences of stubs in Typeshed, forces MyPy to
+# assume that PTransform class is of type Any. Thus to avoid MyPy's error
+# (Class cannot subclass 'PTransform' (has type 'Any')), we added an
+# ignore here.
+class GetVoiceArtistMetadataModelsFromDict(beam.DoFn): # type: ignore[misc]
+    """DoFn to convert voice artist metadata dicts to their corresponding
+    model instances.
+    """
+
+    total_commit_log_models = 0
+    processed_commit_log_models = 0
+
+    @staticmethod
+    def _convert_to_models(
+        voice_artist_id_to_metadata_mapping: Dict[
+            str, voiceover_models.VoiceoversAndContentsMappingType]
+    ) -> List[voiceover_models.VoiceArtistMetadataModel]:
+        """The method converts the voice artist metadata dicts to their
+        corresponding model instances.
+
+        Args:
+            voice_artist_id_to_metadata_mapping:
+                dict(str, VoiceoversAndContentsMappingType). A dictionary
+                mapping voice artist IDs as keys and nested dicts as values.
+                Each nested dict refers to the voiceovers_and_contents_mapping
+                field in the VoiceArtistMetadataModel.
+
+        Returns:
+            list(VoiceArtistMetadataModel). A list of VoiceArtistMetadataModel
+            instances.
+        """
+
+        voice_artist_metadata_models = []
+
+        with datastore_services.get_ndb_context():
+            for voice_artist_id, metadata_mapping in (
+                    voice_artist_id_to_metadata_mapping.items()):
+                voice_artist_model = (
+                    voiceover_services.
+                    create_voice_artist_metadata_model_instance(
+                        voice_artist_id, metadata_mapping)
+                )
+                voice_artist_metadata_models.append(voice_artist_model)
+
+        return voice_artist_metadata_models
+
+    def process(
+        self,
+        voice_artist_id_to_metadata_mapping: Dict[
+            str, voiceover_models.VoiceoversAndContentsMappingType],
+        exp_commit_log_models: List[exp_models.ExplorationCommitLogEntryModel]
+    ) -> Iterable[voiceover_models.VoiceArtistMetadataModel]:
+        self.total_commit_log_models = len(exp_commit_log_models)
```

Comment:
If all you want is the length, then why pass the entire list of models here rather than just the length?

Reply:
removed